### ISSUE TITLE
Nested headers don't update dynamically / move

### ISSUE THREAD

**Issue by allanelder** [handsontable/handsontable#4150]
Hi

I'm evaluating Handsontable PRO for a project; the project requires that I have 2 features.

1. The ability to add columns dynamically
2. The ability to drag / drop columns

### Steps to reproduce

1.Open the codepen
2.Click "Update Cols", the cols don't update, but the correct cols are output to the console.
3.Drag column C over to column B; the data moves, but the column label stays the same.

### Demo
http://codepen.io/robroy72/pen/GWGJdy

### Your environment
* Handsontable version: 1.10.1
* Browser Name and version: Chrome 56.0.2924.87
* Operating System: Windows 7

**Comment by allanelder** [comment]
Update: I tried this through ngHandsontable; the behavior is slightly different, the additional columns are added to the end of the grid but are ungrouped and have no labels,

**Comment by AMBudnik** [comment]
Hi @allanelder 

I am able to change column headers by updateSettings but you are right with moving. We cannot move columns with their names in the 1-lvl headers.

Thank you for sharing.

**Comment by allanelder** [comment]
@AMBudnik hi; just to confirm I understand, you see the new columns being added when you run the codepen and click the button? It doesn't work for me.

**Comment by AMBudnik** [comment]
Yes, I can see that we are missing `{"label":"D","colspan":4}` However we need to create a new ticket for it. It is always better to part issues if it's possible.

**Comment by allanelder** [comment]
@AMBudnik I've updated this existing one for the moving of columns, and added the following for not being able to add nested columns.

https://github.com/handsontable/handsontable/issues/4152

**Comment by AMBudnik** [comment]
Thank you @allanelder

**Comment by AMBudnik** [comment]
The issue is replicable in v `2.0.0.` and doesn't seem to work fine in any of the previous versions.

**Comment by AMBudnik** [comment]
The issue is still replicable in v`5.0.1`. Updated demo https://codepen.io/anon/pen/dqXRRd?editors=1000

![xcgif](https://user-images.githubusercontent.com/10757813/44712718-b6f14a80-aab1-11e8-9a8d-f9275085391a.gif)

**Comment by AMBudnik** [comment]
Replicable in 7.4.2 http://jsfiddle.net/v9d50zhn/
and 8.0.0-beta 1 http://jsfiddle.net/v9d50zhn/1/

**Comment by AMBudnik** [comment]
### Case 2
Moving headers with a different structure
Demo https://jsfiddle.net/AMBudnik/q47s6an5/1/ (8.0.0.-beta2-rev2 still works the same).
![case2](https://user-images.githubusercontent.com/10757813/82306235-10548800-99bf-11ea-8abc-9bcd6e8f189a.gif)

**Comment by pnowak** [comment]
This issue is still replicable using v8.0.0. Here's an updated demo - https://jsfiddle.net/a982xzvd/

**Comment by adrianszymanski89** [comment]
Inform ZD 29841

**Comment by medjedovicm** [comment]
The same issue occurred to me also by triggering move with `manualColumnMove` plugin.

Here is the fiddle: https://jsfiddle.net/medjedovicm/ov9emrzq/17/

**Comment by adrianszymanski89** [comment]
Hi @medjedovicm 

Thank you for the additional information.

**Comment by samsullivan** [comment]
> The same issue occurred to me also by triggering move with `manualColumnMove` plugin. Version: 12.3

I just noticed the same in 13.1

**Comment by adrianszymanski89** [comment]
Hi @samsullivan 

Thank you for pointing that out. This issue isn't on our roadmap yet. We will update this thread once it's fixed.

**Comment by budnix** [comment]
Good news — this issue has been fixed and the solution is ready to be merged (https://github.com/handsontable/handsontable/pull/12793).

Once merged, the fix will be included in the next Handsontable releases. Thank you for reporting the issue.
